operator/pkg/controllers/controller.go

```diff
@@ -106,3 +119,34 @@ func (c *GenericController) reconcile(ctx context.Context, resource Object, pers
 	}
 	return *result, nil
 }
+
+func (c *GenericController) finalizeSubResources(ctx context.Context, resource Object, finalizers []string) (*reconcile.Result, error) {
+	result, err := c.Controller.Finalize(ctx, resource)
+	if err != nil {
+		return result, fmt.Errorf("finalizing resource controller %v, %w", c.Controller.Name(), err)
+	}
+	// Remove finalizer for this controller
+	resource.SetFinalizers(finalizers)
+	zap.S().Infof("[%s] Successfully deleted resources", resource.GetName())
+	return result, nil
+}
+
+func ttlExpiredForControlPlane(resource Object) bool {
+	cp, ok := resource.(*v1alpha1.ControlPlane)
+	if !ok {
+		return false
+	}
+	if cp.Spec.TTL != "" {
```

Comment:
I was wondering can we have some sane defaults ? 
I'm not sure if we can expect customers set this value by default and we might end up with these resources ?

Reply:
Setting default value is tough since we will change the existing functionality and it might lead to surprises if this field is not set in the CP spec. Some use cases like CI  testing require 24 hr TTL other like performance test might require like 7 days retention. Open to ideas.

Reply:
Hmmm, I understand. Given it's in alpha/beta phase, may be we can take a stance now ?
For example - IIRC, default provisioners of karpenter has a very aggressive TTL set today.
Given idea of these guest clusters is that, it is ephemeral by its nature. We can be conservative and start with 10days and document in README ? wdyt ?

Reply:
Ideally we should extend it for some period of time when the cluster is not "idle". e.g. CPU > 20%. But this makes it harder to implement.

I guess we can add a default for TTL, given we are still pre-GA.
This can potentially lead to surprises, we should add it to release notes and README.

Reply:
> This can potentially lead to surprises, we should add it to release notes and README.

Scenario I am worried about is, lets say someone has a created guest cluster in the past and left it running for few days, cleaned up manually. With default TTL now the cluster will get cleaned up automatically which can be a surprise for someone say StormForge for example.

Reply:
Also can we add the default behavior in another PR? Trying to get the 1.25 cluster creation rolled out as well in this PR